tcf_website/management/commands/load_grades.py

```diff
@@ -0,0 +1,398 @@
+import os
+import pandas as pd
+import re
+
+from django.core.management.base import BaseCommand, CommandError
+from tcf_website.models import *
+from django.core.exceptions import ObjectDoesNotExist, MultipleObjectsReturned
+
+from tqdm import tqdm
+
+
+class Command(BaseCommand):
+    """
+    How To Use: Run python3 manage.py load_grades to load all grades
+    in the tcf_website/management/commands/grade_data/excel/ directory.
+    This should take ~20 min to run to load all the grades
+    """
+    help = 'Imports grade data from VAGrades Excel Spreadsheets into PostgresSQL database'
+
+    global course_grades
+    global course_instructor_grades
+
+    course_grades = {}
+    course_instructor_grades = {}
+
+    '''
+    def add_arguments(self, parser):
+
+        # Named (optional) arguments
+        parser.add_argument(
+            '--verbose',
+            action='store_true',
+            help='Verbose output',
+        )
+
+        parser.add_argument(
+            'semester',
+            help='Semester to update grades (e.g. "2019_FALL").\nIf you wish to reload all semesters (potentially dangerous!) then put "ALL_DANGEROUS" as the value of this argument.',
+            type=str
+        )
+    '''
+
+    def handle(self, *args, **options):
+
+        # self.verbose = options['verbose']
+
+        self.data_dir = 'tcf_website/management/commands/grade_data/excel/'
+
+        # semester = options['semester']
+
+        [self.load_semester_file(file) for file in sorted(
+            os.listdir(self.data_dir)) if not file.startswith('.')]
+
+        '''
+        if semester == 'ALL_DANGEROUS':
+            # ignores temporary files
+            [self.load_semester_file(file) for file in sorted(os.listdir(self.data_dir)) if not file.startswith('.')]
+        else:
+            self.load_semester_file(f"{semester.lower()}.xlsx")
+        '''
+
+    def clean(self, df):
+        CourseGrade.objects.all().delete()  # deletes CourseGrade table
+        # deletes CourseInstructorGrade table
+        CourseInstructorGrade.objects.all().delete()
+        return df.dropna(
+            how="all",
+            subset=[
+                'A+',
+                'A',
+                'A-',
+                'B+',
+                'B',
+                'B-',
+                'C+',
+                'C',
+                'C-',
+                'D+',
+                'D',
+                'D-',
+                'F'])
```

Comment:
Seems like this could be compressed a few lines...

Reply:
This is just a matter of preference. I will make the changes, but to play devil's advocate, it's easier to count the number of total lines, the index of each line, and every addition/deletion translates to addition/deletion of a single line (which looks better in `git diff`s).